### PR TITLE
Improvements on contracts ownership

### PR DESCRIPTION
**Pull request by andresaiello**
### Summary
- Add Ownable2Step to Interactor
- Add safeCreate2AndTransfer to create2 factory
- Add test

### CODE REVIEW

File: packages/protocol-contracts/test/ZetaInteractor.spec.ts [hunk 1]
```diff
@@ -50,24 +50,46 @@ describe("ZetaInteractor tests", () => {
...
-      ).to.be.revertedWith(getCustomErrorMessage("InvalidCaller", [deployer.address]));
+      zetaInteractorMock.onZetaRevert({
+        destinationAddress: ethers.utils.solidityPack(["address"], [crossChainContractB.address]),
+        destinationChainId: chainBId,
+        message: encoder.encode(["address"], [zetaInteractorMock.address]),
+        remainingZetaValue: 0,
+        sourceChainId: chainAId,
+        zetaTxSenderAddress: deployer.address
+      });
+    });
```
Comment (lucas-janon) [inline packages/protocol-contracts/test/ZetaInteractor.spec.ts:69]: Why we removed the assertion?

File: packages/protocol-contracts/contracts/ZetaInteractor.sol [hunk 2]
```diff
@@ -1,12 +1,12 @@
 // SPDX-License-Identifier: MIT
 pragma solidity 0.8.7;
 
-import "@openzeppelin/contracts/access/Ownable.sol";
+import "./openzeppelin/Ownable2Step.sol";
```
Comment (lucas-janon) [inline packages/protocol-contracts/contracts/ZetaInteractor.sol:4]: Why not from the dependencies?

Comment (andresaiello) [reply packages/protocol-contracts/contracts/ZetaInteractor.sol:4]: auditors ask us to include it but is not yet in the pkg, it's only in the OZ repo

File: packages/protocol-contracts/test/ZetaInteractor.spec.ts (hunk 1 above)
Comment (andresaiello) [reply packages/protocol-contracts/test/ZetaInteractor.spec.ts]: restored

**Review by lucas-janon — APPROVED** [review]
LGTM

**Pull request merged**